apps/heft/src/plugins/ApiExtractorPlugin/ApiExtractorRunner.ts

```diff
@@ -0,0 +1,144 @@
+
+export interface IApiExtractorRunnerConfiguration {
+  configFileLocation: string;
+  apiExtractorPackagePath: string;
+  typescriptPackagePath: string;
+  buildFolder: string;
+  production: boolean;
+}
+
+export class ApiExtractorRunner extends SubprocessRunnerBase<IApiExtractorRunnerConfiguration> {
+  private _terminal: Terminal;
+
+  public static getTerminal(terminalProvider: ITerminalProvider): Terminal {
+    const prefixTerminalProvider: PrefixProxyTerminalProvider = new PrefixProxyTerminalProvider(
+      terminalProvider,
+      '[api-extractor] '
+    );
+
+    return new Terminal(prefixTerminalProvider);
+  }
+
+  public get filename(): string {
+    return __filename;
+  }
+
+  public initializeTerminal(terminalProvider: ITerminalProvider): void {
+    this._terminal = ApiExtractorRunner.getTerminal(terminalProvider);
+  }
+
+  public async invokeAsync(): Promise<void> {
+    const apiExtractor: typeof TApiExtractor = require(this._configuration.apiExtractorPackagePath);
+    const extractorConfig: TApiExtractor.ExtractorConfig = apiExtractor.ExtractorConfig.loadFileAndPrepare(
+      this._configuration.configFileLocation
+    );
+    const extractorOptions: TApiExtractor.IExtractorInvokeOptions = {
+      localBuild: !this._configuration.production,
+      typescriptCompilerFolder: this._configuration.typescriptPackagePath,
+      messageCallback: (message: TApiExtractor.ExtractorMessage) => {
+        switch (message.logLevel) {
+          case apiExtractor.ExtractorLogLevel.Error: {
+            let logMessage: (IColorableSequence | string)[];
+            if (message.sourceFilePath) {
+              const buildFolderRelativeFilePath: string = path.relative(
+                this._configuration.buildFolder,
+                message.sourceFilePath
+              );
+              logMessage = [
+                Colors.red(
+                  `${buildFolderRelativeFilePath}(${message.sourceFileLine},${message.sourceFileColumn})`
+                ),
+                Colors.white(' - '),
+                Colors.gray(`(${message.category})`),
+                ' ',
+                Colors.red(message.text)
+              ];
+            } else {
+              logMessage = [message.text];
+            }
+
+            this._terminal.writeErrorLine(...logMessage);
+            break;
+          }
+
+          case apiExtractor.ExtractorLogLevel.Warning: {
+            let logMessage: (IColorableSequence | string)[];
+            if (message.sourceFilePath) {
+              const buildFolderRelativeFilePath: string = path.relative(
+                this._configuration.buildFolder,
+                message.sourceFilePath
+              );
+              logMessage = [
+                Colors.yellow(
+                  `${buildFolderRelativeFilePath}(${message.sourceFileLine},${message.sourceFileColumn})`
+                ),
+                Colors.white(' - '),
+                Colors.gray(`(${message.category})`),
```

Comment:
Instead of `Colors.gray`, should it be unformatted?

As far as coloring, we should:
- Try to be conservative, to avoid the unprofessional "Christmas tree" look of some tools
- Make sure it looks good with PowerShell, which has a wildly different color scheme for its console; I spent some time tuning Rush's colors based on that